data-products/src/main/scala/org/sunbird/analytics/audit/AssessmentScoreCorrectionJob.scala

```diff
@@ -0,0 +1,200 @@
+  }
+
+  def saveLocal(data: DataFrame,
+                batchId: String,
+                contentId: String,
+                folderName: String)(implicit config: JobConfig): Unit = {
+    JobLogger.log("Generating the CSV File", Option(Map("batch_id" -> batchId, "content_id" -> contentId)), INFO)
+    val outputPath = config.modelParams.getOrElse(Map[String, Option[AnyRef]]()).getOrElse("csvPath", "").asInstanceOf[String]
+    data.repartition(1).write.option("header", value = true).format("com.databricks.spark.csv").save(outputPath.concat(s"/$folderName-$batchId-$contentId-${System.currentTimeMillis()}.csv"))
+  }
+
+
+  def removeAssessmentRecords(incorrectRecords: DataFrame,
+                              batchId: String,
+                              contentId: String,
+                              isDryRunMode: Boolean)(implicit spark: SparkSession, fc: FrameworkContext, config: JobConfig, sc: SparkContext): Unit = {
+    if (isDryRunMode) {
+      saveLocal(incorrectRecords, batchId, contentId, "assessment-invalid-attempts-records")
+    } else {
+      JobLogger.log("Deleting the records from the table", None, INFO)
+      saveLocal(incorrectRecords, batchId, contentId, "assessment-invalid-attempts-records") // For cross verification purpose
+      incorrectRecords.select("course_id", "batch_id", "user_id", "content_id", "attempt_id").rdd.deleteFromCassandra(AppConf.getConfig("sunbird.courses.keyspace"), "assessment_aggregator", keyColumns = SomeColumns("course_id", "batch_id", "user_id", "content_id", "attempt_id"))
+    }
+  }
+
+  def generateInstructionEvents(inCorrectRecordsDF: DataFrame,
+                                batchId: String,
+                                contentId: String)(implicit spark: SparkSession, fc: FrameworkContext, config: JobConfig, sc: SparkContext): Unit = {
+    val outputPath = config.modelParams.getOrElse(Map[String, Option[AnyRef]]()).getOrElse("csvPath", "").asInstanceOf[String]
+    val file = new File(outputPath.concat(s"/instruction-events-$batchId-$contentId-${System.currentTimeMillis()}.json"))
+    var writer: BufferedWriter = null
+    try {
+      writer = new BufferedWriter(new FileWriter(file))
+      val userIds: List[String] = inCorrectRecordsDF.select("user_id").distinct().collect().map(_ (0)).toList.asInstanceOf[List[String]]
+      val courseId: String = inCorrectRecordsDF.select("course_id").distinct().head.getString(0)
+      for (userId <- userIds) yield {
+        val event = s"""{"assessmentTs":${System.currentTimeMillis()},"batchId":"$batchId","courseId":"$courseId","userId":"$userId","contentId":"$contentId"}"""
+        writer.write(event)
+        writer.newLine()
+        writer.flush()
+      }
+    } catch {
+      case ex: IOException => ex.printStackTrace()
+    } finally {
+      writer.close()
+    }
+  }
+
+  // Method to fetch the totalQuestion count from the content meta
+  def getTotalQuestions(contentId: String,
```

Comment:
Why is the getTotalQuestions encompassed inside a Future? We are making a blocking call to the getTotalQuestions inside the process function and hence a Future is not required.